src/core/UniformsGroup.js

```diff
@@ -0,0 +1,88 @@
+import { EventDispatcher } from './EventDispatcher.js';
+
+/**
+ * @author Mugen87 / https://github.com/Mugen87
+ */
+
+var id = 0;
+
+function UniformsGroup() {
+
+	Object.defineProperty( this, 'id', { value: id ++ } );
+
+	this.name = '';
+
+	this.dynamic = false;
+	this.uniforms = [];
+
+}
+
+UniformsGroup.prototype = Object.assign( Object.create( EventDispatcher.prototype ), {
+
+	constructor: UniformsGroup,
+
+	isUniformsGroup: true,
+
+	add: function ( uniform ) {
+
+		this.uniforms.push( uniform );
+
+		return this;
+
+	},
+
+	remove: function ( uniform ) {
+
+		var index = this.uniforms.indexOf( uniform );
+
+		if ( index !== - 1 ) this.uniforms.splice( index, 1 );
+
+		return this;
+
+	},
+
+	setName: function ( name ) {
+
+		this.name = name;
+
+		return this;
+
+	},
```

Comment:
Do you think some users may want to change the name runtime?

If I understand correctly name is required. And if we can think it's immutable, what do you think of taking name argument as constructor parameter and removing `.setName()`?

```javascript
var group = new THREE.UniformsGroup( 'ViewData' );
```

Reply:
> Do you think some users may want to change the name runtime?

Maybe, although that would actually not make sense. It's like adding uniforms at runtime which would also be an invalid operation. Still, I think the current interface is cleaner than defining everything as ctor parameters.

I consider UBOs as an advanced feature so the API does not need to be foolproof from my point of view^^